File: src/concurrency/elixir_core/omni_toolemu_safety.ex
# Omni ToolEmu Safety Actor (Elixir)
defmodule Omni.ToolEmuSafety do
  @dangerous %{"delete" => 0.9, "write" => 0.6, "execute" => 0.8, "send" => 0.5}
  def assess(action, args) do
    action_lower = String.downcase(action)
    {score, flags} = Enum.reduce(@dangerous, {0.0, []}, fn {da, w}, {s, f} ->
      if String.contains?(action_lower, da), do: {max(s, w), [da | f]}, else: {s, f}
    end)
    level = cond do; score > 0.7 -> "critical"; score > 0.4 -> "high"; true -> "low"; end
    %{score: Float.round(score, 4), level: level, flags: flags}
  end
end

File: src/system/rust_core/omni_toolemu_safety.rs
// Omni ToolEmu Safety Gateway (Rust)
// Ref: ryoungj/ToolEmu — Apache-2.0
pub fn assess_risk(action: &str, args: &[&str]) -> (f64, Vec<String>) {
    let mut score = 0.0f64; let mut flags = Vec::new();
    let dangerous = [("delete", 0.9), ("write", 0.6), ("execute", 0.8), ("send", 0.5)];
    let action_l = action.to_lowercase();
    for (da, w) in &dangerous {
        if action_l.contains(da) { score = score.max(*w); flags.push(format!("action_{}", da)); }
    }
    let sensitive = ["password", "token", "key", "secret"];
    for arg in args {
        let a = arg.to_lowercase();
        for s in &sensitive { if a.contains(s) { score = (score + 0.3).min(1.0); flags.push(format!("sensitive_{}", s)); } }
    }
    (score, flags)
}
